### ISSUE TITLE
Throw error in lfc when attempting to compile reactor without main

### ISSUE THREAD

**Issue by lhstrh** [icyphy/lingua-franca#35]
This is currently not happening. Instead, when executing the generate C code, an error like this is thrown, which is not very descriptive of the problem:
```
/usr/bin/ld: /tmp/ccf537PH.o: in function `initialize':
ImportedC.c:(.text+0x15bf): undefined reference to `__initialize_trigger_objects'
/usr/bin/ld: /tmp/ccf537PH.o: in function `next':
ImportedC.c:(.text+0x1901): undefined reference to `__start_time_step'
/usr/bin/ld: /tmp/ccf537PH.o: in function `main':
ImportedC.c:(.text+0x1cdc): undefined reference to `__start_timers'
collect2: error: ld returned 1 exit status
```

**Comment by edwardalee** [comment]
A main reactor is not required.

Edward

------------
Edward A Lee
Professor
UC Berkeley

On Sep 27, 2019, at 10:37 AM, Marten Lohstroh <notifications@github.com>
wrote:

This is currently not happening. Instead, when executing the generate C
code, an error like this is thrown, which is not very descriptive of the
problem:

/usr/bin/ld: /tmp/ccf537PH.o: in function `initialize':
ImportedC.c:(.text+0x15bf): undefined reference to
`__initialize_trigger_objects'
/usr/bin/ld: /tmp/ccf537PH.o: in function `next':
ImportedC.c:(.text+0x1901): undefined reference to `__start_time_step'
/usr/bin/ld: /tmp/ccf537PH.o: in function `main':
ImportedC.c:(.text+0x1cdc): undefined reference to `__start_timers'
collect2: error: ld returned 1 exit status

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/icyphy/lingua-franca/issues/35?email_source=notifications&email_token=ACA6ONXCLKT6JALQULDTERTQLZADZA5CNFSM4I3I77O2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4HOGYL2Q>,
or mute the thread

**Comment by lhstrh** [comment]
Not for a file that you import elsewhere, but those you also don't compile down to the target. For those that you do compile down to the target, you need main. That is, unless we'll do dynamic linking, but so far we haven't introduced any such machinery. For `lfc`, the command line compiler that generates an executable, it makes total sense to report an error when main is missing.

**Comment by lhstrh** [comment]
Fixed in 09e2de14242975f97f65f8b4a36b8086d5437591.
